### PR TITLE
Added max_drawn_objects argument

### PR DESCRIPTION
**Pull request by big-yellow-duck**
If the max_drawn_objects arguement is not 0, the number of drawn objects can be limited by removing the oldest drawn objects in map

### CODE REVIEW

**Comment by blackary** [comment]
Seems like a nice addition, thanks! Would you be interested in trying to add a test showing this in action on the draw examples page? You can look in test_frontentend.py for examples. If not, I can try and add one at some point.

**Comment by randyzwitch** [comment]
Please also remove any files that are not part of the actual PR

File: streamlit_folium/frontend/package.json [hunk 1]
```diff
@@ -8,12 +8,11 @@
     "@testing-library/user-event": "^7.1.2",
     "@types/hoist-non-react-statics": "^3.3.1",
     "@types/jest": "^24.0.0",
-    "@types/leaflet": "^1.7.9",
     "@types/node": "^12.0.0",
     "bootstrap": "^4.4.1",
     "event-target-shim": "^5.0.1",
     "hoist-non-react-statics": "^3.3.2",
-    "leaflet": "^1.7.1",
+    "leaflet": "^1.9.4",
```
Comment (randyzwitch) [inline streamlit_folium/frontend/package.json:15]: @blackary or @hansthen...with the dynamic insertion of the JS libraries that we now do, do we even need to have leaflet and its peers as part of the dependencies?

**Comment by big-yellow-duck** [comment]
I'll clean it up and make a page for it

File: streamlit_folium/frontend/package.json (hunk 1 above)
Comment (hansthen) [reply streamlit_folium/frontend/package.json:15]: My guess is they can be safely removed. I did a quick test building without these dependencies, and running the tests. The `test_layer_control_dynamic_update` passed only sometimes :-(. I don't think that should be related to the dependencies though. If you want I can make a PR for the changes.

**Comment by big-yellow-duck** [comment]
I updated the PR. but I haven't updated the tests yet. But it's working through by running the example by hand in the draw support page

File: streamlit_folium/frontend/package.json (hunk 1 above)
Comment (randyzwitch) [reply streamlit_folium/frontend/package.json:15]: No need to make a PR, just trying to lean on your conceptual knowledge of what comes out of Folium :)

**Comment by randyzwitch** [comment]
Latest failure due to whitespace in pre-commit check @big-yellow-duck 

https://github.com/randyzwitch/streamlit-folium/actions/runs/9912015159/job/27460574420?pr=202

**Pull request closed without merging**